scripts/sonic_sku_create.py

```diff
@@ -0,0 +1,853 @@
+        "4x10":  { "lanes":4, "speed":10000,  "step":1, "bko":1, "name": "etp" },
+}
+
+bko_dict = bko_dict_4
+
+class SkuCreate(object):
+        """
+	Process aclstat
+	"""
+
+	def __init__(self):
+
+		self.portconfig_dict = {}
+		self.platform_specific_dict = {"x86_64-mlnx_msn2700-r0":self.msn2700_specific}
+		self.default_lanes_per_port = []
+                self.platform = None
+                self.base_lanes = None
+		self.fpp = []
+		self.fpp_split  =  {}
+		self.num_of_fpp = 0
+		self.num_of_lanes_dict = {"1" : 4, "2" : 2, "4" : 1}
+		self.sku_name = None
+                self.default_sku_path = None
+		self.base_sku_name = None
+		self.base_sku_dir = None
+		self.base_file_path = None		
+		self.new_sku_dir = None
+		self.print_mode = False
+		self.remove_mode = False
+		self.verbose = None
+
+	def read_metadata(self):
+		"""
+		Redis database platform info
+		"""
+		try:
+			self.metadata = self.db.get_all(self.db.CONFIG_DB, "DEVICE_METADATA|localhost" )
+		except:
+			print ("Error While trying to retrieve METADATA from CONFIG_DB", file=sys.stderr)
+			exit(1)
+			
+	def sku_def_parser(self,sku_def) :
+		# Parsing XML sku definition file to extract Interface speed and InterfaceName(alias) <etp<#><a/b/c/d> to be used to analyze split configuration
+		# Rest of the fields are used as placeholders for portconfig_dict [name,lanes,SPEED,ALIAS,index]
+		try:
+			f = open(str(sku_def),"r")
+		except IOError:
+			print ("Couldn't open file: " + str(sku_def), file=sys.stderr)
+			exit(1)
+		element = ET.parse(f)
+		
+		root = element.getroot()
+		if (self.verbose):
+			print ( "tag=%s, attrib=%s" % (root.tag, root.attrib))
+		self.sku_name = root.attrib["HwSku"]
+		self.new_sku_dir = self.default_sku_path+"/"+self.sku_name+ '/'
+		idx = 1
+		for child in root:
+			if child.tag == "Ethernet":
+				for interface in child:
+					for eth_iter in interface.iter():
+						if eth_iter is not None:
+							self.portconfig_dict[idx] = ["Ethernet"+str(idx),[1,2,3,4], eth_iter.get("InterfaceName"),  str(idx), eth_iter.get("Speed")]
+							if (self.verbose):
+								print ("sku_def_parser:portconfig_dict[",idx,"] -> ",self.portconfig_dict[idx])
+							idx += 1
+
+		f.close()
+
+	def parse_deviceinfo(self,meta,hwsku):
+		# Parsing minigraph sku definition file to extract Interface speed and InterfaceName(alias) <etp<#><a/b/c/d> to be used to analyze split configuration
+		# Rest of the fields are used as placeholders for portconfig_dict [name,lanes,SPEED,ALIAS,index]
+		idx = 1
+		match = None
+		for device_info in meta.findall(str(QName(minigraph_ns, "DeviceInfo"))):
+			dev_sku = device_info.find(str(QName(minigraph_ns, "HwSku"))).text
+			if dev_sku == hwsku :
+				match = True
+				interfaces = device_info.find(str(QName(minigraph_ns, "EthernetInterfaces"))).findall(str(QName(minigraph_ns1, "EthernetInterface")))
+				for interface in interfaces:
+					alias = interface.find(str(QName(minigraph_ns, "InterfaceName"))).text
```

Comment:
@madhanmellanox please fix all indentations - 4 spaces. Also please align codestyle with PEP8.

Reply:
Addressed in next commit.